encord/client.py

```diff
@@ -504,10 +506,32 @@ def add_private_data_to_dataset(
         else:
             raise ValueError(f"Type [{type(private_files)}] of argument private_files is not supported")
 
-        payload = {"files": files, "integration_id": integration_id, "ignore_errors": ignore_errors}
-        response = self._querier.basic_setter(DatasetData, self._config.resource_id, payload=payload)
+        process_hash = self._querier.basic_setter(
+            DatasetDataLongPolling,
+            self._config.resource_id,
+            payload={
+                "files": files,
+                "integration_id": integration_id,
+                "ignore_errors": ignore_errors,
+            },
+        )["process_hash"]
+
+        for _ in range(10):
+            polling_response = self._querier.basic_getter(
```

Comment:
If we have a shorter timeout this can actually be much nicer as it will not trigger our classic SDK timeout errors. Let's say if you re-request every 10 seconds.

Reply:
That's specified in BE